src/recipes/heartbeat-by-user-first-impression/utils.js

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*jshint forin:true, noarg:false, noempty:true, eqeqeq:true, bitwise:true,
+  strict:true, undef:true, curly:false, browser:true,
+  unused:true,
+  indent:2, maxerr:50, devel:true, node:true, boss:true, white:true,
+  globalstrict:true, nomen:false, newcap:true, esnext: true, moz: true  */
+
+/*global require, exports, log */
+
+
+let cutBreaks = function (arr, breaks, rng=Math.random()) {
+  // should have tests, gah!  was not ready to deal with this yet.
+  // should have way of overring rng too
+  let out;
+  for (let ii = 0; ii < breaks.length; ii++) {
+    if (rng <= breaks[ii]) {
+      return arr[ii];
+    }
+  }
+};
+
+var getEngagementUrl = function(locale, opts) {
+  let ans;
+  if (locale == "de") {
+    return null;
+  }
+  if (/^en-/.test(locale)) {
+    let ans = cutBreaks(surveyUrls['en'], surveyBreaks['en'])
```

Comment:
Shouldn't surveyUrls and surveyBreaks be parameters now that getEngagementUrl is in its own file?